systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/objects/PodUtils.java

```diff
@@ -238,6 +238,61 @@ public static void waitUntilPodLabelsDeletion(String namespaceName, String podNa
         }
     }
 
+    /**
+     * Waits until the single specified annotation of a Pod changes its value.
+     *
+     * @param namespaceName    the namespace in which the Pod is located.
+     * @param podName          the name of the Pod whose annotation value needs to be checked.
+     * @param annotationKey    the key of the annotation whose value needs to be checked.
+     * @param annotationValue  the expected value of the annotation to change from.
+     *
+     * @throws TimeoutException if the annotation value does not change within the {@code DELETION_TIMEOUT} period.
+     */
+    public static void waitUntilPodAnnotationChange(String namespaceName, String podName, String annotationKey, String annotationValue) {
+
+        LOGGER.info("Waiting for Pod: {}/{} to change annotation value: {}-{}", namespaceName, podName, annotationKey, annotationValue);
+        TestUtils.waitFor("Pod to change value of annotation" + annotationKey + " with key " + annotationKey, Constants.POLL_INTERVAL_FOR_RESOURCE_READINESS,
+            DELETION_TIMEOUT, () ->
+                !annotationValue.equals(kubeClient(namespaceName).getPod(namespaceName, podName).getMetadata().getAnnotations().get(annotationKey))
+        );
+        LOGGER.info("Pod: {}/{} changed annotation value: {}-{}", namespaceName, podName, annotationKey, annotationValue);
+    }
+
+    /**
+     * Waits for a specific annotation key-value pair to remain stable (from start to beginning) on a given Pod.
+     *
+     * @param namespaceName   the namespace in which the Pod is located.
+     * @param annotationKey   the key of the annotation to be checked.
+     * @param annotationValue the expected value of the annotation to be checked.
+     * @param observedPodName the name of the Pod whose annotation is being observed.
+     *
+     * @throws TimeoutException if the annotation does not remain stable within the {@code Constants.GLOBAL_STATUS_TIMEOUT} period.
+     */
+    public static void waitForPrevailedPodAnnotationKeyValuePairs(String namespaceName, String annotationKey, String annotationValue, String observedPodName) {
```

Comment:
I guess you don't need to have the same thing on two places -> in `PodUtils` you can have this waits and in `StrimziPodSetUtils` you can have some encapsulation, which will pass the Pod names to the `PodUtils` methods.

Reply:
actually i can't as I am writing in the the PR description, if i would look into Pod  (belonging to Kafka Node Pool Controller role, and i would change some `spec.kafka.config` this should cause changes in annotation for config hash, but this annotation is  visibly changed only when we get to this Pod inside SPS CR, not inside Pod as such (i.e. (kubectl get pod my-controller-pod would still have the old value of mentioned annotation) (I think that changing this annotation would actually cause rolling of given pod)

Reply:
Okay, thanks for the explanation :)